.travis.yml

```diff
@@ -1,35 +1,46 @@
-branches:
-    only:
-        - master
-language: python
-python:
-    - "3.8"
-services:
-    - docker
-matrix:
+branches: master
+language: minimal
+git: { submodules: false }
+jobs:
     include:
-        - language: python
-          name: "Documentation"
+        - stage: "Lint"
+          name: "General Lints"
+          script:
+            - "! git grep -n '^<<<<<<< '"  # Check for Git conflict markers
+
+        - name: "Python Lints"
+          language: python
           python: 3.8
           cache: pip
+          install:
+            - >-
+              pip install --disable-pip-version-check
+              black flake8 flake8-import-order
+          script:
+            - flake8 kuma docs tests
+            - black --check --diff kuma docs tests
+
+        - name: "JavaScript Lints"
+          language: node_js
+          node_js: 12
+          cache: npm
           script:
-            - scripts/ci-docs
-        - language: python
-          name: "Python linting"
+            - scripts/ci-jslint
+            - scripts/ci-flow
+
+        - stage: "Test"
+          name: "Documentation Build"
+          language: python
           python: 3.8
           cache: pip
+          install:
+            - pip install -r docs/requirements.txt
           script:
-            - set -x
-            # Make sure there are no left-over git patch markers in
-            - git --no-pager grep '>>>>>>>'
-            - pip install poetry~=1.0.0b9
-            - poetry --version
-            - poetry install -vv
```

Comment:
The reason it used `poetry` was so we would get the same exact version of `black`, `flake8`, and `flake8-import-order` (and all the dependencies locked) as we have in `pyproject.toml` was so that you get the same linting effect in CI as you'd get if you run `docker-compose exec web flake8 kuma` (for example). 

I guess it's rather immaterial and if it's ever an issue we can add `pip install flake8==VERSION ...` and manually maintain those to match what's inside the `pyproject.toml`. 

Then again, does it really matter?

Reply:
I agree that being absolutely certain that we use the same versions everywhere is ideal, but I'm willing to relax that for linting tools. Plus, we can also achieve that ideal by committing to keeping our main repo current: If things start failing because Travis is using a newer version of Black, then it's time to update the poetry.lock! 😉 

Honestly, I want to revisit this, but right now the time savings are too great to ignore.

What I'd love to do, eventually, is find a way to run `poetry install` _once_ in Travis to pre-cache everything (e.g., https://docs.travis-ci.com/user/build-stages/warm-cache/) and then re-use that cache for the linting, pytest, etc. work. But since pytest is in Docker things get complicated. I think I can find a way to make it work, but it'll take more research.

Reply:
I did try, many months ago, based on some googling of mentioned hacks, get Travis to cache the docker image stuff. That way, instead of caching the pip and npm caches, it would just cache the docker image in some smart way. I did some decent looking attempts but I managed to "crash" Travis to the point where it was just unresponsive and no way to debug why it's not building anything. 

I think in my mind I thought "Man, that used to be so much easier in CircleCI that I don't wanna bother".